mantidproject/mantid
# Add extra features to the debug logging for MC absorption calculation

**Description of work.**

Add these two new features to the debug logging generated when the Monte Carlo scattering point generation and track generation is run:
- Statistics on the scattering angle (mean and standard deviation)
- Separately log the number of scattering point generated and used (was previously just the count generated)

This is mainly of interest on elastic instruments. The change was motivated by the work done on the absorption weighted path length calculation for WISH where we may in future wish to fix the scattering angle for all generated tracks for a particular peak

**To test:**

Run the following script which will trigger the new logging code in MCInteractionVolume. If the logging level is set to debug you should see logging information showing the mean and standard deviation of the scattering angle for the tracks generated while processing each peak in the input workspace:

```
from mantid.simpleapi import *
import matplotlib.pyplot as plt
import numpy as np

peaks = LoadIsawPeaks(Filename=r'Peaks5637.integrate')

# create cylindrical sample with radius=1cm
SetSample(peaks,Geometry={'Shape': 'Cylinder','Height': 5.0,'Radius': 1.0,'Center': [0.,0.,0.]}, 
                          Material={'ChemicalFormula': 'V'})

# populate the t bar column in the peaks workspace                         
AddAbsorptionWeightedPathLengths(peaks,1000)
```

Fixes #28733 .

*This does not require release notes* because PR #28703 (new AddAbsorptionWeightedPathLengths algorithm) already has general entry in the Diffraction release notes and this PR is an extra detail relating to that new algorithm

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
